[NO-JIRA] Alert GH Action Failures on Slack

belen-pruvost-sonarsource opened the pull request:
Similar to https://github.com/SonarSource/docker-sonarqube/pull/543

I'd like to make sure that we are notified on [#team-sq-devops](https://sonarsource.slack.com/archives/C0372LQ1N06) for whenever a GH action of this repo fails.

I don't have access to set up secrets on this repo. @tobias-trabelsi-sonarsource could you set that? The webhook url could be that of the [App](https://api.slack.com/apps/A0376E7S685/incoming-webhooks?) we created for this.

Side note: does anyone besides you @tobias-trabelsi-sonarsource have access to set up secrets for this repo?

tobias-trabelsi-sonarsource commented:
@belen-pruvost-sonarsource i don't have access to manage secrets on GH. Please send a request to the Release Engineering team for that

tobias-trabelsi-sonarsource commented on .github/workflows/hadolint-analysis.yml:
```diff
@@ -17,3 +17,10 @@ jobs:
       run: docker pull hadolint/hadolint
     - name: Lint Dockerfile
       run: docker run --rm --interactive hadolint/hadolint hadolint --ignore DL3018 --ignore DL4001 --ignore DL3013 --ignore SC2015 - < ./4/Dockerfile
+    - name: Notify failures on Slack
```
i would remove this for a quality gate that will only trigger on PR and needs to pass prior to being able to merge. a dedicated slack notification does not seem necessary

belen-pruvost-sonarsource replied:
It makes sense, I'll remove it, thanks!

tobias-trabelsi-sonarsource commented on .github/workflows/rebuild.yml:
```diff
@@ -21,6 +21,13 @@ jobs:
           image-ref: 'docker.io/sonarsource/sonar-scanner-cli'
           exit-code: 1
           ignore-unfixed: true
+      - name: Notify failures on Slack
```
I would remove this slack notification here as well. the condition that there is a CVE in the image will trigger a rebuild of the image with updated upstream packages that will hopefully fix this CVE. for vulnerabilities in the application itself we should look at the sonar-scanner-cli repository.
getting a notification whenever there is a upstream package with a vulnerability that will automatically updated during the same run seems not necessary for me

tobias-trabelsi-sonarsource commented on .github/workflows/rebuild.yml:
```diff
@@ -45,6 +52,14 @@ jobs:
         else
           echo "::set-output name=changed::false"
         fi
+    - name: Notify failures on Slack
```
a notification that we could not check the hash of a new base image implies that somebody updated the base image without taking the rebuild condition into consideration. it is arguably if this should trigger a slack notification or not

tobias-trabelsi-sonarsource commented on .github/workflows/rebuild.yml:
```diff
@@ -98,4 +113,11 @@ jobs:
-          docker push repox-sonarsource-docker-releases.jfrog.io/sonarsource/sonar-scanner-cli:${{ steps.get_version.outputs.major }}
\ No newline at end of file
+          docker push repox-sonarsource-docker-releases.jfrog.io/sonarsource/sonar-scanner-cli:${{ steps.get_version.outputs.major }}
+      - name: Notify failures on Slack
```
This makes sense as it would take the whole rebuild pipeline into consideration :+1:

belen-pruvost-sonarsource commented:
I've realized that `secrets.SLACK_WEBHOOK` is probably pointing to #team-sonarqube-build, so before addressing your comments I'll reach out to the RE team to ask them to create a secret for a webhook url that points to #team-sq-devops instead
